frontend/search.js

```diff
@@ -0,0 +1,78 @@
+const searchBar = document.getElementById('keywordSearch');
+
+// Debounce function to limit the rate at which the search function executes
+function debounce(func, delay) {
+    let debounceTimer;
+    return function () {
+        const context = this;
+        const args = arguments;
+        clearTimeout(debounceTimer);
+        debounceTimer = setTimeout(() => func.apply(context, args), delay);
+    };
+}
+
+//Levenshtein distance function, calculates number of edits needed to make
```

Comment:
I don't understand this, can this comment be updated to better describe the purpose of this function?